language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -319,12 +319,12 @@ impl LibraVM {
         remote_cache: &StateViewCache<'_>,
         writeset_payload: &WriteSetPayload,
         txn_sender: Option<AccountAddress>,
-    ) -> Result<ChangeSet, Result<(VMStatus, TransactionOutput), VMStatus>> {
+    ) -> Result<(ChangeSet, bool), Result<(VMStatus, TransactionOutput), VMStatus>> {
```

Comment:
Nit: instead of returning an extra bool value, this code might be more readable if we add 
```needs_reconfig(&self): bool {
  match self {
    WriteSetPayload::Direct(_) => true,
    ... 
  }
}```
 to `impl WriteSetPayload` + use that at line 472 instead?

Reply:
I like it! I'll update